### PR TITLE
Renamed DFGraph to GraphFrames

### PR DESCRIPTION
**Pull request by jkbradley**
Specifically, renamed:
* dfgraph -> graphframes (package)
* DFGraph -> GraphFrame (class)

Tests pass, and docs compile, though there are issues generating Python API docs (which are an issue apparently in DFGraph, though they compiled at some point previously).

CC: @thunterdb Can you please review?
CC: @ankurdave FYI  Did you want to list the names of the primary contributors to this project somewhere?  I changed it to say Berkeley, MIT, and Databricks, but am open to changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,12 @@
-#spark-df-graph
+#graphframes
 
-# DFGraph: DataFrame-based Graphs
+# GraphFrames: DataFrame-based Graphs
```
Comment (thunterdb) [inline README.md]: I will never get used to the Capitalization of DataFrame...

File: README.md [hunk 2]
```diff
@@ -29,3 +24,8 @@ You will need to set `SPARK_HOME` to your local Spark installation directory.
 This project is compatible with Spark 1.4+.  However, significant speed improvements have been
 made to DataFrames in more recent versions of Spark, so you may see speedups from using the latest
 Spark version.
+
+## Contributing
+
+GraphFrames is collaborative effort among UC Berkeley, MIT, and Databricks.
+We welcome open source contributions as well!
```
Comment (thunterdb) [inline README.md]: Do you want to put a link to Spark's contribution guide?

**Comment by thunterdb** [comment]
LGTM, with a tiny comment that we can address later.

**Comment by jkbradley** [comment]
OK thanks!  I'll merge it.

**Pull request merged**